pywincffi/core/typesbase.py

```diff
@@ -0,0 +1,58 @@
+"""
+Types Base
+==========
+
+Provides the base types on top of which user visible types will be built.
+"""
+
+import cffi
+
+
+# pylint: disable=too-few-public-methods
+class CFFICDataWrapper(object):
+
+    """
+    Base class for exposing Python types and interfaces to pywincffi users.
+    * Wraps a CFFI cdata object in self._cdata.
+    * Delegates attribute getting/setting to self._cdata, supporting structs.
+    * Delegates item getting/setting to self._cdata, supporting arrays.
+    Attribute access is not delegated to the wrapped object if the class
+    itself contains such an attribute and that attribute is a descriptor; this
+    is in place to support @property in sub-classes.
```

Comment:
Making ffi a non-optional arg. Benefit: Ensure loose coupling between this class and the remaining code.

Notes:
* wintypes was already providing the dist.load() ffi instance.
* test_typesbase with minor review, explicitly passing "dummy" ffi instance.